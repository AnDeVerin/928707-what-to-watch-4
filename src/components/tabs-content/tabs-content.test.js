import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import NameSpace from '../../reducer/name-space.js';

import TabsContent from './tabs-content.jsx';

const mockStore = configureStore([]);

const movie = {
  id: 1,
  title: `Fantastic Beasts: The Crimes of Grindelwald`,
  thumbUrl: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  genre: `Adventure`,
  realeseYear: 2018,
  posterUrl: `https://m.media-amazon.com/images/M/MV5BYWVlMDI5N2UtZTIyMC00NjZkLWI5Y2QtODM5NGE5MzA0NmVjXkEyXkFqcGdeQXVyNzU3NjUxMzE@._V1_.jpg`,
  coverUrl: `https://m.media-amazon.com/images/M/MV5BYTRiOTM4YzEtN2MwMC00MjMxLWIxM2ItMGE1YzA1MjNkN2YzXkEyXkFqcGdeQXVyNjQ4ODE4MzQ@._V1_SX1777_CR0,0,1777,999_AL_.jpg`,
  trailer: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  overview: {
    rating: {
      value: 6.6,
      count: 240,
    },
    description: `At the end of the first film, the powerful Dark wizard Gellert Grindelwald (Depp) was captured by MACUSA (Magical Congress of the United States of America), with the help of Newt Scamander (Redmayne). But, making good on his threat, Grindelwald escaped custody and has set about gathering followers, most unsuspecting of his true agenda: to raise pure-blood wizards up to rule over all non-magical beings. In an effort to thwart Grindelwald's plans, Albus Dumbledore (Law) enlists his former student Newt Scamander, who agrees to help, unaware of the dangers that lie ahead. Lines are drawn as love and loyalty are tested, even among the truest friends and family, in an increasingly divided wizarding world.`,

    director: `David Yates`,
    stars: [
      `Johnny Depp`,
      `Eddie Redmayne`,
      `Kevin Guthrie`,
      `Carmen Ejogo`,
      `Wolf Roth`,
      `Eddie Redmayne`,
      `ZoÃ« Kravitz`,
      `Callum Turner`,
      `Derek Riddell`,
      `Cornell John`,
      `Ezra Miller`,
    ],
    runTime: 134,
  },
};

const review = {
  id: 1,
  user: {
    id: 4,
    name: 'Kate Muir',
  },
  rating: 8.9,
  comment:
    "Discerning travellers and Wes Anderson fans will luxuriate in the glorious Mittel-European kitsch of one of the director's funniest and most exquisitely designed movies in years.",
  date: '2019-05-08T14:13:56.569Z',
};

describe('TabsContent component', () => {
  it('renders correctly on active tab Overview', () => {
    const component = renderer
      .create(<TabsContent movie={movie} activeTab={`overview`} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('renders correctly on active tab Details', () => {
    const component = renderer
      .create(<TabsContent movie={movie} activeTab={`details`} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  xit('renders correctly on active tab Reviews', () => {
    const store = mockStore({
      [NameSpace.DATA]: {
        reviews: [review],
      },
    });

    const component = renderer
      .create(
        <Provider store={store}>
          <TabsContent movie={movie} activeTab={`reviews`} />
        </Provider>
      )
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  it('renders nothing on unknown active tab', () => {
    const component = renderer
      .create(<TabsContent movie={movie} activeTab={`some unknown tab`} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });
});
