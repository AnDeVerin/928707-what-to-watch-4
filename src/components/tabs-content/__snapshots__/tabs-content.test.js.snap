// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsContent component renders correctly on active tab Details 1`] = `
<div
  className="movie-card__text movie-card__row"
>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Director
      </strong>
      <span
        className="movie-card__details-value"
      >
        David Yates
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Starring
      </strong>
      <span
        className="movie-card__details-value"
      >
        <span>
          Johnny Depp
           
          <br />
        </span>
        <span>
          Eddie Redmayne
           
          <br />
        </span>
        <span>
          Kevin Guthrie
           
          <br />
        </span>
        <span>
          Carmen Ejogo
           
          <br />
        </span>
        <span>
          Wolf Roth
           
          <br />
        </span>
        <span>
          Eddie Redmayne
           
          <br />
        </span>
        <span>
          ZoÃ« Kravitz
           
          <br />
        </span>
        <span>
          Callum Turner
           
          <br />
        </span>
        <span>
          Derek Riddell
           
          <br />
        </span>
        <span>
          Cornell John
           
          <br />
        </span>
        <span>
          Ezra Miller
           
          <br />
        </span>
      </span>
    </p>
  </div>
  <div
    className="movie-card__text-col"
  >
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Run Time
      </strong>
      <span
        className="movie-card__details-value"
      >
        2h 14m
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Genre
      </strong>
      <span
        className="movie-card__details-value"
      >
        Adventure
      </span>
    </p>
    <p
      className="movie-card__details-item"
    >
      <strong
        className="movie-card__details-name"
      >
        Released
      </strong>
      <span
        className="movie-card__details-value"
      >
        2018
      </span>
    </p>
  </div>
</div>
`;

exports[`TabsContent component renders correctly on active tab Overview 1`] = `
Array [
  <div
    className="movie-rating"
  >
    <div
      className="movie-rating__score"
    >
      6.6
    </div>
    <p
      className="movie-rating__meta"
    >
      <span
        className="movie-rating__level"
      >
        Good
      </span>
      <span
        className="movie-rating__count"
      >
        240
         ratings
      </span>
    </p>
  </div>,
  <div
    className="movie-card__text"
  >
    <p>
      At the end of the first film, the powerful Dark wizard Gellert Grindelwald (Depp) was captured by MACUSA (Magical Congress of the United States of America), with the help of Newt Scamander (Redmayne). But, making good on his threat, Grindelwald escaped custody and has set about gathering followers, most unsuspecting of his true agenda: to raise pure-blood wizards up to rule over all non-magical beings. In an effort to thwart Grindelwald's plans, Albus Dumbledore (Law) enlists his former student Newt Scamander, who agrees to help, unaware of the dangers that lie ahead. Lines are drawn as love and loyalty are tested, even among the truest friends and family, in an increasingly divided wizarding world.
    </p>
    <p
      className="movie-card__director"
    >
      <strong>
        Director: 
        David Yates
      </strong>
    </p>
    <p
      className="movie-card__starring"
    >
      <strong>
        Starring: 
        Johnny Depp, Eddie Redmayne, Kevin Guthrie
         and other
      </strong>
    </p>
  </div>,
]
`;

exports[`TabsContent component renders nothing on unknown active tab 1`] = `null`;
